server/internal/core/file/file.go

```diff
@@ -0,0 +1,64 @@
+package file
+
+import (
+	"context"
+	"errors"
+	"io"
+	"path/filepath"
+
+	"github.com/google/uuid"
+	"github.com/oxidrive/oxidrive/server/internal/core/user"
+	"github.com/rs/zerolog"
+)
+
+type FileID (uuid.UUID)
+type Content io.Reader
+type Path string
+type Name string
+type Size int
+
+func (i FileID) String() string {
+	return uuid.UUID(i).String()
+}
+
+var (
+	ErrInvalidPath = errors.New("the provided file path is invalid")
+)
+
+type File struct {
+	ID      FileID
+	Content Content
+	Name    Name
+	Path    Path
+	Size    Size
+}
+
+func NewFile(content Content, path Path, size Size) (*File, error) {
```

Comment:
```suggestion
func Create(content Content, path Path, size Size) (*File, error) {
```
Naming nitpick: I prefer more expressive names to "constructor-style" functions. `file.Create` reads better that `file.NewFile`, and `file.New` doesn't convey the feeling of "action" like `Create` does.

Reply:
`Create` is ambiguous, multiple structs (file and service in this case) are defined in the same package

Reply:
Even if the module name is the same as the struct? It's pretty unambiguous that `file.Create` creates a file, while `file.NewService` initializes a new FileService
Ultimately it's no big deal :)

Reply:
Good point 🤔 

What about this?
```
struct File { ... }
struct FileService { ... }

func Create() File { ... }
func InitService() File { ... }
```
and the final result when you use the `file` package would be
```
import ".../.../.../file"

uploaded := file.Create(...)
fileService := file.InitService(...)
fileService.Upload(uploaded)
```

Reply:
Yeah that works. I'm not hot on `Init` vs `New` for the service but I'm not against either, as long as we keep all services similar. If you decide to go for `InitService` please rename the UserService as well